### DISCUSSION TITLE
Merge Queue Checks issues

### DISCUSSION THREAD

**Discussion by kyle-hiller-nh** [community/community#37368] [Pull Requests]
I'm trying to implement a merge queue that runs a deployment and integration tests against the new deployment. I have some issues that occur when PRs in the queue get batched together.

1) Relying on a required status that fails if integration tests fail

Whenever this occurs and PRs are batched together (i.e. 3 PRs are in the queue, one has checks running, the other two are waiting) the original set of checks for PR#2 are canceled and replaced with a grouped set of checks for PR#2 and PR#3. 

Once the checks for PR#1 pass, the PR is merged correctly. However, after the grouped checks for PR#2 and PR#3 are run, the checks for PR#3 are marked as passed but not PR#2 so neither PR is merged and the whole queue is basically frozen and needs to be cleared out and checks re-run. 

Basically this means that we can have at most 1 running PR and one waiting PR in the queue or else nothing gets merged.

2) Relying on successful deployment to environment to merge.

Similarly to above, we gate a "success" status for deployment on the passing of integration tests. However, when the deployment is marked as success, the PR is not merged and is instead added to a group in the Merge Queue that says "next up for deployment" with the subheading "waiting for merge group to be locked". I'm unsure what locking the merge queue is, and once the queue of checks is drained still nothing gets merged and they sit in this group forever.

Has anyone experienced these issues/have a workaround for them?

**Comment by ossanna16** [comment]
Hi there @kyle-hiller-nh and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
